### PR TITLE
retry realtime on error

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

- Automatically reconnect realtime connection

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

* #108 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

**Review by stnguyen90 — COMMENTED** [review]
How did you test this? I tried using android emulator and turning on and off airplane mode, but it seemed like the onDone and onError was never called.

File: lib/src/realtime_mixin.dart [hunk 1]
```diff
@@ -87,34 +92,45 @@ mixin RealtimeMixin {
...
+  }
+
+  int _getTimeout() {
+    return _retries < 5
+        ? 1
+        : _retries < 15
+            ? 5
+            : _retries < 100
+                ? 10
+                : 60;
```
[the remark this answers is not in the document]
Comment (lohanidamodar) [reply lib/src/realtime_mixin.dart:131]: This is better.

**Comment by lohanidamodar** [comment]
> How did you test this? I tried using android emulator and turning on and off airplane mode, but it seemed like the onDone and onError was never called.

That's exactly what I did, turn off internet, wait for few seconds and turn on again, it works for me

**Review by stnguyen90 — COMMENTED** [review]
hmm so in dartpad the switch expression works and the docs say you just need dart 3.0 for it to work, but it isn't working locally for me on dart 3.3: 
image
did you see this?

**Comment by lohanidamodar** [comment]
> hmm so in dartpad the switch expression works and the docs say you just need dart 3.0 for it to work, but it isn't working locally for me on dart 3.3: image did you see this?

I'm sure it's because of this
https://github.com/appwrite/sdk-for-flutter/blob/f40d4cbbaa767cfffa60b72b95f467fc4fcbc21f/pubspec.yaml#L16

So I will revert it back, as we want it to be compatible with old Dart versions as long as possible.

**Comment by stnguyen90** [comment]
> turn off internet

Can you clarify what you did exactly so I can reproduce? Did you test on an emulator or physical device?

**Review by stnguyen90 — COMMENTED** [review]
> turn off internet

Can you clarify what you did exactly so I can reproduce? Did you test on an emulator or physical device? I still can't reproduce. Here's a screenshot of airplane mode on, a log entry for a successful subscription, but no log entry for disconnect/retry:
image

**Comment by lohanidamodar** [comment]
> > turn off internet
> 
> Can you clarify what you did exactly so I can reproduce? Did you test on an emulator or physical device? I still can't reproduce. Here's a screenshot of airplane mode on, a log entry for a successful subscription, but no log entry for disconnect/retry: image

Here's what I did on emulator
1. Connect and subscribe
2. Disconnect Wifi/Internet (Didn't try Airplane mode, but it should be same)
3. Wait for few seconds
4. Open wifi again
5. Create document
6. You should see the realtime log update

**Review by stnguyen90 — APPROVED** [review]
@lohanidamodar, hmm it's working now 🤷🏼

**Comment by puntiz** [comment]
Was this fix for realtime reconnect not merged ?

**Comment by lohanidamodar** [comment]
> Was this fix for realtime reconnect not merged ?

This is just a draft PR for testing. The fix is already merged and released.

**Pull request closed without merging**